packages/editor/src/components/rich-text/index.js

```diff
@@ -436,13 +438,16 @@ export class RichText extends Component {
 	 *                                the live DOM.
 	 */
 	onChange( record, _withoutApply ) {
+		// Filter out annotations
+		const filteredRecord = removeFormat( record, 'mark', 0, record.text.length );
```

Comment:
We should make sure the right APIs are in place so we don't have to modify `RichText` at all. Working on that after the Formatting API.

Reply:
I'll need some more information first to understand what you want to do, which I should have asked before. Why is there formatting be stripped on every change? That seems a bit odd. I can understand that you don't want the format to be serialised, but then we should leave it away as part of the serialisation (to HTML). This also makes me think that we might benefit from having and "editor only" format, that get automatically stripped out. Maybe it can even be something different: only a node that's put in place when the DOM is created for the editor, just like the way we will make invisible character visible (non breaking space etc.) and might create inline boundaries based on the selection.

Reply:
I opted for not having the formatting for annotations in the state of the editor, because that would mean that they would end up in the attributes of blocks. And I don't think they are actually part of the blocks. The use-cases of marking something up is always a meta concern:

* Spellchecking
* Editorial highlighting
* Highlighting pieces of text for writing assistance

Putting these into the attributes of a block would muddy the actual content of a block.

What I can see is another abstraction underneath `RichText` that handles 'formatting that isn't in the state'. Which annotations could be one use-case of. I don't have any other use-cases for such an abstraction though, so I opted for the simplest solution.